synapse/module_api/__init__.py

```diff
@@ -1340,6 +1349,91 @@ async def store_remote_3pid_association(
         """
         await self._store.add_user_bound_threepid(user_id, medium, address, id_server)
 
+    async def add_push_rule_for_user(
```

Comment:
I think much of this method is copied from `PushRuleRestServlet` -- would it make sense to create a `PushRuleHandler` to encapsulate that logic and call it from both here and `PushRuleRestServlet`? (Parts of that class could probably be moved into the handler too...)

Reply:
Yes probably.